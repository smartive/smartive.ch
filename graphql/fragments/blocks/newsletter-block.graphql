fragment newsletterBlock on NewsletterRecord {
  id
  content {
    value
    blocks {
      ... on LinkRecord {
        _modelApiKey
        ...links
      }
    }
  }
  image {
    id
    title
    responsiveImage(imgixParams: { w: 752, h: 752, auto: format }) {
      ...responsiveImage
    }
  }
}
