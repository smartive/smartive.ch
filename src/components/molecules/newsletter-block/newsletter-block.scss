@import '../../../scss/colors';
@import '../../../scss/responsive';

.newsletter-block {
  margin: 8rem 0 5rem;

  @media #{$large-up} {
    margin: 10rem 0;
  }

  @media #{$xlarge-up} {
    margin: -2rem 0 8rem;
  }
}
