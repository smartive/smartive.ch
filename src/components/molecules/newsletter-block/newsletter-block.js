import React from 'react';

import './newsletter-block.scss';
import { NewsletterSignup } from '../newsletter-signup';

export const NewsletterBlock = () => (
  <div className="container">
    <div className="newsletter-block">
      <div className="row">
        <div className="col-xs-10 col-xs-offset-1    col-sm-8 col-sm-offset-2    col-lg-6 col-lg-offset-3">
          <h4>Neugierig, was bei smartive läuft?</h4>
          <p>
            Unser Newsletter erscheint in lockeren Abständen. Er enthält spannende Infos zu digitalen Trends und Insights zu
            unserem Team.
          </p>
          <NewsletterSignup />
        </div>
      </div>
    </div>
  </div>
);

export default NewsletterBlock;
