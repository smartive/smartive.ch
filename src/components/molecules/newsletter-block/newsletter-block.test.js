import React from 'react';
import renderer from 'react-test-renderer';

import { NewsletterBlock } from './newsletter-block';

describe('Newsletter Block', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NewsletterBlock />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
