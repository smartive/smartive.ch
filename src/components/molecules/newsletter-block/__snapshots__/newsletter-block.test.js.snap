// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter Block renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="newsletter-block"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-10 col-xs-offset-1 col-sm-8 col-sm-offset-2 col-lg-6 col-lg-offset-3"
      >
        <h4>
          Neugierig, was bei smartive läuft?
        </h4>
        <p>
          Unser Newsletter erscheint in lockeren Abständen. Er enthält spannende Infos zu digitalen Trends und Insights zu unserem Team.
        </p>
        <div
          className="newsletter-signup"
        >
          <form
            onSubmit={[Function]}
          >
            <input
              className="input"
              disabled={false}
              name="newsletter-email"
              onChange={[Function]}
              placeholder="Email-Adresse eintragen"
              type="email"
              value=""
            />
            <button
              className="button"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Anmelden
            </button>
            <div
              className="newsletter-signup__info-container "
            />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
